google-cloud-spanner/src/test/java/com/google/cloud/spanner/BenchmarkingUtilityTest.java

```diff
@@ -0,0 +1,190 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.spanner;
+
+import com.google.api.gax.rpc.ServerStream;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.rpc.Code;
+import com.google.spanner.v1.BatchWriteResponse;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Hosts a bunch of utility methods/scripts that can be used while performing benchmarks to load
+ * data, report latency metrics, etc.
+ *
+ * <p>Table schema used here: CREATE TABLE FOO ( id INT64 NOT NULL, BAZ INT64, BAR INT64, )
+ * PRIMARY KEY(id);
+ */
+@Category(SlowTest.class)
+@RunWith(JUnit4.class)
+public class BenchmarkingUtilityTest {
+
+  @ClassRule
+  public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  // TODO(developer): Add your values here for PROJECT_ID, INSTANCE_ID, DATABASE_ID
+  // TODO(developer): By default these values are blank and should be replaced before a run.
+  private static final String PROJECT_ID = "";
+  private static final String INSTANCE_ID = "";
+  private static final String DATABASE_ID = "";
+  private static final String SERVER_URL = "https://staging-wrenchworks.sandbox.googleapis.com";
+  private static DatabaseClient client;
+  private static Spanner spanner;
+
+  @BeforeClass
+  public static void beforeClass() {
+    final SpannerOptions.Builder optionsBuilder =
+        SpannerOptions.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setAutoThrottleAdministrativeRequests();
+    if (!SERVER_URL.isEmpty()) {
+      optionsBuilder.setHost(SERVER_URL);
+    }
+    final SpannerOptions options = optionsBuilder.build();
+    spanner = options.getService();
+    client = spanner.getDatabaseClient(DatabaseId.of(PROJECT_ID, INSTANCE_ID, DATABASE_ID));
+
+    // Delete all existing data from the table
+    client.write(ImmutableList.of(Mutation.delete("FOO", KeySet.all())));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    spanner.close();
+  }
+
+  /**
+   * A utility which bulk inserts 10^5 records into the database in batches. The method assumes that
+   * the instance/database/table is already created. It does not perform any admin operations.
+   *
+   * <p>Table schema used here: CREATE TABLE FOO ( id INT64 NOT NULL, BAZ INT64, BAR INT64, )
+   * PRIMARY KEY(id);
+   */
+  @Test
```

Comment:
Do we want this as an actual test that is executed during (nightly?) builds? If so, it should verify that the test actually succeeds by adding at least one assertion.

Or would it make more sense to move this method to the actual benchmark class, so it can more easily be executed together with a benchmark?

Reply:
I have annotated `BenchmarkingUtilityTest` with `@Category(SlowTest.class)`. This will make sure its excluded from our pre-submits and nightly tests. Note that the data loading utility is a one time activity and need to be executed repeatedly along with the benchmarks.

Hence, I think it will be better to leave this as a standalone class.